File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = list()

    for bracket in s:
        if bracket == '(':
            stack.append(bracket)
        else:
            if not len(stack):
                return False
            else:
                stack.pop()

    if not len(stack):
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <stack>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;

    stack<char> items;
    
    for(int i=0; i<s.length(); i++) {
        if (s[i] == '(') items.push(s[i]);
        if (s[i] == ')') {
            if (!items.empty()) {
                items.pop();
            }
            else {
                answer = false;
                break;
            }
        }
    }
    
    if (!items.empty()) answer = false; 
    
    return answer;
}